src/internal/lokiutil/repair.go

```diff
@@ -0,0 +1,102 @@
+package lokiutil
+
+// This file repairs log lines from broken Loki configurations.  Learn about the log formats here:
+// https://github.com/kubernetes/design-proposals-archive/blob/main/node/kubelet-cri-logging.md
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+func tryDocker(input map[string]any) (result string) {
```

Comment:
Maybe returning an `ok` rather than overloading `result` is better? Can a Docker log entry ever have a blank log? I.e., is `{"log": ""}` a valid Docker log entry?

Reply:
I think it's valid, but not possible.  The logs are split by lines and lines have line terminators which are retained:

```
$ docker run -it -d debian:stable sh -c 'echo; echo; echo'                                                                                                                                                                                        
57abf7b437e7de5ea883d33972961d235e97b4d874330860c564d47614b11ee2  
root@vm:/var/lib/docker/containers/57abf7b437e7de5ea883d33972961d235e97b4d874330860c564d47614b11ee2# cat 57abf7b437e7de5ea883d33972961d235e97b4d874330860c564d47614b11ee2-json.log
{"log":"\r\n","stream":"stdout","time":"2024-07-10T16:58:10.954768889Z"}
{"log":"\r\n","stream":"stdout","time":"2024-07-10T16:58:10.954817591Z"}
{"log":"\r\n","stream":"stdout","time":"2024-07-10T16:58:10.954820091Z"}
```

Reply:
I also answered your original question in the (updated) PR description.  Yes, anyone can `print '{"log":"hi"}` from their app and we will only return `hi`... but we wrote all the apps we're collecting logs from and they don't do that ;)